subsys/net/lib/sockets/getaddrinfo.c

```diff
@@ -103,31 +103,73 @@ static void dns_resolve_cb(enum dns_resolve_status status,
 	state->idx++;
 }
 
+static k_timeout_t recalc_timeout(k_timepoint_t end, k_timeout_t timeout)
+{
+	k_timepoint_t new_timepoint;
+
+	timeout.ticks <<= 1;
+
+	new_timepoint = sys_timepoint_calc(timeout);
+
+	if (sys_timepoint_cmp(end, new_timepoint) < 0) {
+		timeout = sys_timepoint_timeout(end);
+	}
+
+	return timeout;
+}
+
 static int exec_query(const char *host, int family,
 		      struct getaddrinfo_state *ai_state)
 {
 	enum dns_query_type qtype = DNS_QUERY_TYPE_A;
+	k_timepoint_t end = sys_timepoint_calc(K_MSEC(CONFIG_NET_SOCKETS_DNS_TIMEOUT));
+	k_timeout_t timeout;
```

Comment:
It can be set directly?

```suggestion
	k_timeout_t timeout = K_MSEC(MIN(CONFIG_NET_SOCKETS_DNS_TIMEOUT,
					 CONFIG_NET_SOCKETS_DNS_BACKOFF_INTERVAL));
```

Reply:
Sure, will fix.